static/app/views/settings/project/server-side-sampling/modals/uniformRateModal.tsx

```diff
@@ -125,7 +126,7 @@ export function UniformRateModal({
     );
 
     setActiveStep(
-      clientDiscard || !isProjectOnOldSDK
+      (clientDiscard || !isProjectOnOldSDK) && !isSamplingSdkVersionsBeingProcessed()
```

Comment:
Why do we need to check `isSamplingSdkVersionsBeingProcessed` and show the step 0 modal?

Step 0 modal should be displayed only if you don't have client outcomes (fetched from the org stats endpoint)

Reply:
yes, I think there was a misunderstanding when we discussed this last week... will update the code in a minute.

One think I noticed is that when the "processing" is done and the project is incompatible, we don't let the users add a uniform rule, but when we don't have yet this information we do... shouldn't we remove this check only don't let users activate a uniform rule if the project is incompatible?

Reply:
here is where we have part of the logic https://github.com/getsentry/sentry/blob/717b4e740dcc7a6a40263ce3a1fbad1bf87ea592/static/app/views/settings/project/server-side-sampling/utils/useRecommendedSdkUpgrades.tsx#L22

Reply:
I think we can keep it as is, that's fine 👍